### PR TITLE
[2018.3] minion_blackout for scheduled jobs

### PR DESCRIPTION
**Pull request by garethgreenaway**
### What does this PR do?
Adding support for minion_blackout for jobs that are run from the Salt scheduler.

### What issues does this PR fix or reference?
#50562 

### Tests written?
Yes

### Commits signed with GPG?
Yes

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/latest/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/utils/schedule.py [hunk 1]
```diff
@@ -713,7 +734,6 @@ def handle_func(self, multiprocessing_enabled, func, data):
 
                 ret['success'] = True
 
-                data_returner = data.get('returner', None)
```
Comment (cachedout) [inline salt/utils/schedule.py]: I'm not following the reasoning for this change. Help? :)

Comment (garethgreenaway) [reply salt/utils/schedule.py]: @cachedout Definitely.  I moved it outside the try...except.  When an exception happens, eg. the new one I added for minion_blackout or if an invalid call happens, the code will drop into the `finally` section where data_returner is referenced.  In some situations it hasn't been created yet, so errors appear in the logs.

**Review by rares-pop — APPROVED** [review]
LGTM

File: salt/utils/schedule.py (hunk 1 above)
Comment (cachedout) [reply salt/utils/schedule.py]: Makes perfect sense. Thank you!

**Review by cachedout — APPROVED** [review]

**Pull request merged**